diff --git a/node_modules/react-native-version-check-expo/src/ExpoVersionInfo.js b/node_modules/react-native-version-check-expo/src/ExpoVersionInfo.js
index 03fd474..6e2de79 100644
--- a/node_modules/react-native-version-check-expo/src/ExpoVersionInfo.js
+++ b/node_modules/react-native-version-check-expo/src/ExpoVersionInfo.js
@@ -11,9 +11,7 @@ if (process.env.RNVC_ENV === 'test') {
     currentVersion: '0.0.1',
   };
 } else {
-  const manifest = Constants.manifest
-    ? Constants.manifest
-    : Constants.manifest2.extra.expoClient;
+  const manifest = Constants.expoConfig;
   const {
     version = null,
     android: { versionCode = null, package: androidPackageName = null } = {},
