diff --git a/node_modules/react-native-swipe-gestures/index.js b/node_modules/react-native-swipe-gestures/index.js
index 38d1db5..fbb83a0 100644
--- a/node_modules/react-native-swipe-gestures/index.js
+++ b/node_modules/react-native-swipe-gestures/index.js
@@ -58,8 +58,8 @@ class GestureRecognizer extends Component {
 
   _gestureIsClick(gestureState) {
     return (
-      Math.abs(gestureState.dx) < swipeConfig.gestureIsClickThreshold &&
-      Math.abs(gestureState.dy) < swipeConfig.gestureIsClickThreshold
+      Math.abs(gestureState.dx) < swipeConfig.gestureIsClickThreshold
+      // && Math.abs(gestureState.dy) < swipeConfig.gestureIsClickThreshold
     );
   }
 
